flag_string_slice.go

```diff
@@ -132,7 +132,51 @@ func lookupStringSlice(name string, set *flag.FlagSet) []string {
 		if err != nil {
 			return nil
 		}
+		// extract default value from the flag
+		var defaultVal []string
+		for _, v := range strings.Split(f.DefValue, ",") {
+			defaultVal = append(defaultVal, v)
+		}
+		// if the current value is not equal to the default value
+		// remove the default values from the flag
+		if !isStringSliceEqual(parsed, defaultVal) {
+			for _, v := range defaultVal {
+				parsed = removeFromStringSlice(parsed, v)
+			}
+		}
 		return parsed
 	}
 	return nil
 }
+
+func removeFromStringSlice(slice []string, val string) (newVal []string) {
+	var count int
+	for _, v := range slice {
+		if v == val {
+			newVal = slice[count+1:]
+			return
+		}
+		newVal = append(newVal, v)
+		count++
+	}
+	return
+}
+
+func isStringSliceEqual(newValue, defaultValue []string) bool {
+	// If one is nil, the other must also be nil.
+	if (newValue == nil) != (defaultValue == nil) {
+		return false
+	}
+
+	if len(newValue) != len(defaultValue) {
+		return false
+	}
+
+	for i, v := range newValue {
+		if v != defaultValue[i] {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
Request: Make sure to run `gofmt` on these files. 

Long term, this is something we should enforce in the pipeline as well.

Reply:
> Long term, this is something we should enforce in the pipeline as well.

=> https://github.com/urfave/cli/issues/993

Reply:
This is done